bin/main.rb

```diff
@@ -101,24 +91,28 @@ def game_board(board_tokens) # rubocop:todo Metrics/CyclomaticComplexity
 
   sleep(1)
 
-  game_board(board_tokens)
+  tokens.game_board
 
   if breakpoint.zero?
     puts 'the game ended in a draw'
     break
   end
 
-  if board_1 == [1, 2, 3] or board_1 == [4, 5, 6] or board_1 == [7, 8, 9] or board_1 == [3, 5, 7] or board_1 == [1, 5, 9] or board_1 == [1, 4, 7] or board_1 == [2, 5, 8] or board_1 == [3, 6, 9] or board_2 == [3, 2, 1]
-    puts "#{player_1}, you are the winner!"
+  first_player = Logic.new(board_1)
```

Comment:
Also the simple fact to pass `board` as a parameter in three classes (Position, ArrayOperation and Logic) is a prove of an error into your OOP implementation

As indicated above, merge the three class into one class called `Board` and all their method must now be the methods of that new `Board` class.

Dont forget to add an attribute `attr_reader :board`.